### DISCUSSION TITLE
Login with Mastodon

### DISCUSSION THREAD

**Discussion by erlend-sh** [ory/hydra#3571] [Ideas]
Mastodon supports OAuth login: https://docs.joinmastodon.org/spec/oauth/

Can Ory enable my app to let users login with their Mastodon accounts?

**Comment by kmherrmann** [comment]
Hydra itself acts as an OAuth provider that delegates login. 
Ory Identities and Ory Kratos should be able to use Mastodon to authenticate users if you set it up as a generic OIDC provider and Mastodon implements the spec without any custom quirks.

**Reply by vinckr** [reply]
@erlend-sh 

Please let me know how it goes. 
Would be great to have a short tutorial on "How to implement Login with Mastodon with Ory"!
